### ISSUE TITLE
New lint: qualified `None` to help make code more readable

### ISSUE THREAD

**Issue by cloneable** [rust-lang/rust-clippy#14523]
### What it does

I'd like to suggest a new lint that checks for naked `None`s and asks to qualify them with the inner type of the `Option`.

### Advantage

- Code becomes more readable, especially when looked at without inlay hints.
- Might warn against unexpected type changes, that a naked `None` wouldn't catch.

### Drawbacks

- This may not be for everyone, so I suggest "restricted" as the target lint group.

### Example

```rust
fn myfunc(s: Option<String>) { .. }
```

```rust
myfunc(None);
```

Could be written as:

```rust
myfunc(None::<String>);
```

The lint should also ask to rewrite instances of `Option::<T>::None` as `None::<T>`.
